### PR TITLE
Save the sasl state in the session

### PR DESCRIPTION
**Pull request by LemonBoy**

### CODE REVIEW

File: src/irc/core/irc-session.c [hunk 1]
```diff
@@ -65,6 +67,10 @@ static void sig_session_save_server(IRC_SERVER_REC *server, CONFIG_REC *config,
 	config_node_set_str(config, node, "away_reason", server->away_reason);
 	config_node_set_bool(config, node, "emode_known", server->emode_known);
 
+	config_node_set_int(config, node, "sasl_mechanism", server->connrec->sasl_mechanism);
+	config_node_set_str(config, node, "sasl_password", server->connrec->sasl_password);
+	config_node_set_str(config, node, "sasl_password", server->connrec->sasl_password);
```
Comment (ahf) [inline src/irc/core/irc-session.c]: Twice?

File: src/irc/core/irc-session.c [hunk 2]
```diff
@@ -90,6 +96,10 @@ static void sig_session_restore_server(IRC_SERVER_REC *server,
 	server->emode_known = config_node_get_bool(node, "emode_known", FALSE);
 	server->isupport_sent = config_node_get_bool(node, "isupport_sent", FALSE);
 
+	server->connrec->sasl_mechanism = config_node_get_int(node, "sasl_mechanism", SASL_MECHANISM_NONE);
+	server->connrec->sasl_password = g_strdup(config_node_get_str(node, "sasl_password", NULL));
+	server->connrec->sasl_password = g_strdup(config_node_get_str(node, "sasl_password", NULL));
```
Comment (ahf) [inline src/irc/core/irc-session.c]: Same.

File: src/irc/core/irc-session.c [hunk 3]
```diff
@@ -90,6 +96,10 @@ static void sig_session_restore_server(IRC_SERVER_REC *server,
 	server->emode_known = config_node_get_bool(node, "emode_known", FALSE);
 	server->isupport_sent = config_node_get_bool(node, "isupport_sent", FALSE);
 
+	server->connrec->sasl_mechanism = config_node_get_int(node, "sasl_mechanism", SASL_MECHANISM_NONE);
+	server->connrec->sasl_username = g_strdup(config_node_get_str(node, "sasl_username", NULL));
+	server->connrec->sasl_password = g_strdup(config_node_get_str(node, "sasl_password", NULL));
```
Comment (dequis) [inline src/irc/core/irc-session.c]: Strdup with no corresponding free? Maybe overwriting an existing value in connrec? (haven't checked)

**Comment by LemonBoy** [comment]
@ahf 
I'm not very good at doing copy/paste/replace/repeat at late night heh

@dx
It was indeed loading the fields from the config before getting to that point, fixed that.

**Comment by dequis** [comment]
@LemonBoy wrong dx

**Comment by dequis** [comment]
This one is fine too.

**Pull request merged**